### DISCUSSION TITLE
Nested write failing when disconnecting midway

### DISCUSSION THREAD

**Discussion by rhlsthrm** [prisma/prisma#7149] [Q&A] [answered]
We have a nested write in our schema. We have seen data corruption issues and we are investigating why it would be happening. We wrote a bunch of tests and discovered if we disconnect Prisma midway through the nested write we can reproduce data corruption issues due to partial saves.

We are using Prisma in a mission-critical production application so this is a huge problem for us. Has anyone experienced this or know what we can do about this?

We are doing a pretty complicated query with multiple nested `connectOrCreate` calls.

**Comment by ryands17** [comment, marked as answer]
@rhlsthrm 👋 

It would be great if you could create an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with all the reproduction steps so that we can look into this and see why this is happening.
